### PR TITLE
Implement `cap_add` and `cap_drop` handling for `docker_swarm_service`

### PR DESCRIPTION
**Pull request by andrasmaroy**
##### SUMMARY
With the [5.0.3 release of Docker SDK](https://github.com/docker/docker-py/releases/tag/5.0.3) for Python support was added for these parameters for swarm services. With that adding support in Ansible should be as simple as exposing these variables for the module. That's what's done in this change.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

docker_swarm_service

### CODE REVIEW

File: tests/integration/targets/docker_swarm_service/tasks/tests/options.yml [hunk 1]
```diff
@@ -1920,3 +1920,79 @@
...
+    state: absent
+  diff: no
+
+- assert:
+    that:
+    - capabilities_1 is changed
+    - capabilities_2 is not changed
+    - capabilities_3 is not changed
+    - capabilities_4 is changed
+  when: docker_py_version is version('5.0.3', '>=')
```
Comment (felixfontein) [inline tests/integration/targets/docker_swarm_service/tasks/tests/options.yml:1991]: You need to check the API version here as well, in the CI failures it has a new enough docker-py, but not a new enough API version.

**Comment by andrasmaroy** [comment]
I'm a bit confused about these CI failures, the errors seem unrelated to me, am I just not seeing the full picture?

**Comment by felixfontein** [comment]
The CI errors are caused by your changes, see my explanation here: https://github.com/ansible-collections/community.docker/pull/294#discussion_r800148713 Capabilities for swarm services need both a new enough Docker SDK for Python version **and** a new enough API version. The former is available on the CI nodes which fail, the latter is not.

File: tests/integration/targets/docker_swarm_service/tasks/tests/options.yml [hunk 2]
```diff
@@ -1920,3 +1920,79 @@
...
+    - capabilities_2 is not changed
+    - capabilities_3 is not changed
+    - capabilities_4 is changed
+  when: docker_api_version is version('1.41', '>=') and docker_py_version is version('5.0.3', '>=')
+
+- assert:
+    that:
+    - capabilities_1 is failed
+    - "('version is ' ~ docker_py_version ~ ' ') in capabilities_1.msg"
+    - "'Minimum version required is 5.0.3 ' in capabilities_1.msg"
```
Comment (felixfontein) [inline tests/integration/targets/docker_swarm_service/tasks/tests/options.yml:1997]: The error message that is failing CI is `Docker API version is 1.40. Minimum version required is 1.41 to set cap_add option.`:
```suggestion
    - >
      (('version is ' ~ docker_py_version ~ ' ') in capabilities_1.msg and 'Minimum version required is 5.0.3 ' in capabilities_1.msg)
      or (('Docker API version is ' ~ docker_api_version ~ '. ') in capabilities_1.msg and 'Minimum version required is 1.41 ' in capabilities_1.msg)
```

**Comment by felixfontein** [comment]
BTW, you also need to mention the two new return values in the `swarm_service` return value example.

File: tests/integration/targets/docker_swarm_service/tasks/tests/options.yml (hunk 2 above)
Comment (andrasmaroy) [reply tests/integration/targets/docker_swarm_service/tasks/tests/options.yml:1997]: Shoot I missed this, thanks for the explanation!

File: plugins/modules/docker_swarm_service.py [hunk 3]
```diff
@@ -1773,6 +1797,10 @@ def compare(self, os):
             force_update = True
         if self.init is not None and self.init != os.init:
             differences.add('init', parameter=self.init, active=os.init)
+        if self.cap_add is not None and self.cap_add != os.cap_add:
```
Comment (felixfontein) [inline plugins/modules/docker_swarm_service.py:1800]: Lists should be compared with `has_list_changed`. It also takes care of sorting (I assume the order of the capabilities does not matter; if it does, add `sort_lists=False`):
```suggestion
        if has_list_changed(self.cap_add, os.cap_add):
```

File: plugins/modules/docker_swarm_service.py [hunk 4]
```diff
@@ -1773,6 +1797,10 @@ def compare(self, os):
...
+            differences.add('cap_add', parameter=self.cap_add, active=os.cap_add)
+        if self.cap_drop is not None and self.cap_drop != os.cap_drop:
```
Comment (felixfontein) [inline plugins/modules/docker_swarm_service.py:1802]:
```suggestion
        if has_list_changed(self.cap_drop, os.cap_drop):
```

**Review by felixfontein — APPROVED** [review]
Looks good to me. I would be glad if someone more familiar with this module could also take a look at this :)

If nobody does, I'll also merge this one in roughly a week (probably next Saturday).

**Comment by WojciechowskiPiotr** [comment]
lgtm

**Comment by felixfontein** [comment]
@andrasmaroy also thanks for this contribution!
@WojciechowskiPiotr thanks again for reviewing!

**Pull request merged**